### ISSUE TITLE
Context Deadline Exceeded

### ISSUE THREAD

**Issue by yousefcodes** [guregu/dynamo#81]
```
func (h Historical) BatchUpdateCustomerData(customerID string) (int, error) {
	fmt.Printf("Updating customer: %s", customerID)
	items := []map[string]*dynamodb.AttributeValue{}
	tbl := h.gDyn.Table(h.table)
	qry := tbl.Get("customer_id", customerID).Range("date", gdynamo.Greater, 0).Index(h.index)
	ct, _ := qry.Count()
	fmt.Sprintf("updating %d itmes", ct)
	qry.All(&items)
	btch := tbl.Batch().Write()
	for _, item := range items {
		btch.Put(item)
	}

	res, err := btch.Run()

	fmt.Sprintf("updated %d itmes", ct)
	if err != nil {
		return res, err
	}

	return res, nil
}
```

The above code executes fine for small data sets. For 4000+ items, I'm getting an error that looks like this:
```
Error: RequestCanceled: request context canceled
caused by: context deadline exceeded
```

**Comment by yousefcodes** [comment]
This is a problem with my autoscaling configuration I believe? It would be nice if we had access to the retry amount. I guess the custom context solves that problem, `RunWithContext`

**Comment by guregu** [comment]
Yes, that is what the context functions are for. You can also change the default value by setting RetryTimeout: https://godoc.org/github.com/guregu/dynamo#RetryTimeout
```go
func init() {
     dynamo.RetryTimeout = 5 * time.Minute
)
```
BTW, in my experience DynamoDB autoscaling takes a while to kick in, at least a few minutes. If you know you're going to be hitting the DB hard it helps to manually scale up beforehand.

**Comment by yousefcodes** [comment]
That is true, I've noticed that! Cool on the `RetryTimeout` accessibility.

**Comment by sushiljacksparrow** [comment]
I am seeing the same problem now. I have a local dynamo db instance running. I am seeing this error when creating a table. Here is my table creation code

```
sess := session.Must(session.NewSession())
db := dynamo.New(sess, aws.NewConfig().WithEndpoint(endpoint).WithRegion("us-east-1").WithCredentials(credentials.NewStaticCredentials("random-1", "random-2", "random-3")))
err := db.CreateTable(tableName, DispatchOrder{}).OnDemand(true).Run()
```

**Comment by guregu** [comment]
@sushiljacksparrow Are you sure you can actually connect to the local instance? You can use `aws.LogDebugWithHTTPBody` to see the requests getting sent. Can you connect with https://github.com/aaronshaf/dynamodb-admin etc? Please confirm it's really an issue with this library.

**Comment by jonathan-kosgei** [comment]
@guregu I'm running into this as well when trying to put 3M+ items into a table.

**Comment by guregu** [comment]
@jonathan-kosgei If you're using methods that don't take context, you can change your timeout as outlined within this issue https://github.com/guregu/dynamo/issues/81#issuecomment-442131576. Or you can use the methods that take context and control it individually.
If you use `context.Background()` it will never time out. You can set custom timeouts with [`context.WithTimeout`](https://pkg.go.dev/context#WithTimeout).
